BENCH_DATA M4_COMP_ECONOMICS <pyaf.Bench.TS_datasets.cTimeSeriesDatasetSpec object at 0x7fcd76b45b38>
TIME :  Date N= 9 H= 6 HEAD= [0 1 2 3 4] TAIL= [4 5 6 7 8]
SIGNAL :  ECON1880 N= 9 H= 6 HEAD= [ 43.73008  53.37139  98.27069  80.37858  88.046  ] TAIL= [  88.046    116.49295  133.2751   297.53     257.38378]
   ECON1880  Date
0  43.73008     0
1  53.37139     1
2  98.27069     2
3  80.37858     3
4  88.04600     4
<class 'pandas.core.frame.DataFrame'>
Int64Index: 9 entries, 0 to 8
Data columns (total 2 columns):
ECON1880    9 non-null float64
Date        9 non-null int64
dtypes: float64(1), int64(1)
memory usage: 216.0 bytes
None
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.5/logging/__init__.py", line 983, in emit
    stream.write(msg)
ValueError: I/O operation on closed file.
Call stack:
  File "/usr/local/bin/ipython3", line 11, in <module>
    sys.exit(start_ipython())
  File "/usr/local/lib/python3.5/dist-packages/IPython/__init__.py", line 119, in start_ipython
    return launch_new_instance(argv=argv, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 595, in launch_instance
    app.initialize(argv)
  File "<decorator-gen-111>", line 2, in initialize
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 74, in catch_config_error
    return method(app, *args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/IPython/terminal/ipapp.py", line 311, in initialize
    self.init_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 298, in init_code
    self._run_cmd_line_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 423, in _run_cmd_line_code
    self._exec_file(fname, shell_futures=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 353, in _exec_file
    raise_exceptions=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/interactiveshell.py", line 2493, in safe_execfile
    self.compile if kw['shell_futures'] else None)
  File "/usr/local/lib/python3.5/dist-packages/IPython/utils/py3compat.py", line 185, in execfile
    exec(compiler(f.read(), fname, 'exec'), glob, loc)
  File "/home/antoine/dev/python/packages/pyaf/benches/run_benchmark_M4Comp.py", line 17, in <module>
    tester.run_multiprocessed();
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 335, in run_multiprocessed
    pool = mp.Pool(processes=nbprocesses, maxtasksperchild=None)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 118, in Pool
    context=self.get_context())
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 168, in __init__
    self._repopulate_pool()
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 233, in _repopulate_pool
    w.start()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 105, in start
    self._popen = self._Popen(self)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 267, in _Popen
    return Popen(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 20, in __init__
    self._launch(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 74, in _launch
    code = process_obj._bootstrap()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 249, in _bootstrap
    self.run()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 119, in worker
    result = (True, func(*args, **kwds))
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 59, in run_bench_process
    tester.testSignal(a.mSignal, a.mHorizon)
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 163, in testSignal
    self.trainSignal(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 158, in trainSignal
    self.trainModel(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 145, in trainModel
    lAutoF1.train(df , self.mTSSpec.mTimeVar , iSignal, iHorizon)
  File "/home/antoine/dev/python/packages/pyaf/ForecastEngine.py", line 25, in train
    self.mSignalDecomposition.train(iInputDS, iTime, iSignal, iHorizon, iExogenousData);
  File "/home/antoine/dev/python/packages/pyaf/TS/SignalDecomposition.py", line 459, in train
    logger.info("START_TRAINING '" + iSignal + "'")
Message: "START_TRAINING 'ECON1880'"
Arguments: ()
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.5/logging/__init__.py", line 983, in emit
    stream.write(msg)
ValueError: I/O operation on closed file.
Call stack:
  File "/usr/local/bin/ipython3", line 11, in <module>
    sys.exit(start_ipython())
  File "/usr/local/lib/python3.5/dist-packages/IPython/__init__.py", line 119, in start_ipython
    return launch_new_instance(argv=argv, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 595, in launch_instance
    app.initialize(argv)
  File "<decorator-gen-111>", line 2, in initialize
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 74, in catch_config_error
    return method(app, *args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/IPython/terminal/ipapp.py", line 311, in initialize
    self.init_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 298, in init_code
    self._run_cmd_line_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 423, in _run_cmd_line_code
    self._exec_file(fname, shell_futures=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 353, in _exec_file
    raise_exceptions=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/interactiveshell.py", line 2493, in safe_execfile
    self.compile if kw['shell_futures'] else None)
  File "/usr/local/lib/python3.5/dist-packages/IPython/utils/py3compat.py", line 185, in execfile
    exec(compiler(f.read(), fname, 'exec'), glob, loc)
  File "/home/antoine/dev/python/packages/pyaf/benches/run_benchmark_M4Comp.py", line 17, in <module>
    tester.run_multiprocessed();
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 335, in run_multiprocessed
    pool = mp.Pool(processes=nbprocesses, maxtasksperchild=None)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 118, in Pool
    context=self.get_context())
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 168, in __init__
    self._repopulate_pool()
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 233, in _repopulate_pool
    w.start()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 105, in start
    self._popen = self._Popen(self)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 267, in _Popen
    return Popen(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 20, in __init__
    self._launch(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 74, in _launch
    code = process_obj._bootstrap()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 249, in _bootstrap
    self.run()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 119, in worker
    result = (True, func(*args, **kwds))
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 59, in run_bench_process
    tester.testSignal(a.mSignal, a.mHorizon)
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 163, in testSignal
    self.trainSignal(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 158, in trainSignal
    self.trainModel(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 145, in trainModel
    lAutoF1.train(df , self.mTSSpec.mTimeVar , iSignal, iHorizon)
  File "/home/antoine/dev/python/packages/pyaf/ForecastEngine.py", line 25, in train
    self.mSignalDecomposition.train(iInputDS, iTime, iSignal, iHorizon, iExogenousData);
  File "/home/antoine/dev/python/packages/pyaf/TS/SignalDecomposition.py", line 481, in train
    logger.info("END_TRAINING_TIME_IN_SECONDS '" + iSignal + "' " + str(self.mTrainingTime))
Message: "END_TRAINING_TIME_IN_SECONDS 'ECON1880' 5.693732023239136"
Arguments: ()
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.5/logging/__init__.py", line 983, in emit
    stream.write(msg)
ValueError: I/O operation on closed file.
Call stack:
  File "/usr/local/bin/ipython3", line 11, in <module>
    sys.exit(start_ipython())
  File "/usr/local/lib/python3.5/dist-packages/IPython/__init__.py", line 119, in start_ipython
    return launch_new_instance(argv=argv, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 595, in launch_instance
    app.initialize(argv)
  File "<decorator-gen-111>", line 2, in initialize
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 74, in catch_config_error
    return method(app, *args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/IPython/terminal/ipapp.py", line 311, in initialize
    self.init_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 298, in init_code
    self._run_cmd_line_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 423, in _run_cmd_line_code
    self._exec_file(fname, shell_futures=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 353, in _exec_file
    raise_exceptions=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/interactiveshell.py", line 2493, in safe_execfile
    self.compile if kw['shell_futures'] else None)
  File "/usr/local/lib/python3.5/dist-packages/IPython/utils/py3compat.py", line 185, in execfile
    exec(compiler(f.read(), fname, 'exec'), glob, loc)
  File "/home/antoine/dev/python/packages/pyaf/benches/run_benchmark_M4Comp.py", line 17, in <module>
    tester.run_multiprocessed();
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 335, in run_multiprocessed
    pool = mp.Pool(processes=nbprocesses, maxtasksperchild=None)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 118, in Pool
    context=self.get_context())
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 168, in __init__
    self._repopulate_pool()
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 233, in _repopulate_pool
    w.start()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 105, in start
    self._popen = self._Popen(self)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 267, in _Popen
    return Popen(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 20, in __init__
    self._launch(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 74, in _launch
    code = process_obj._bootstrap()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 249, in _bootstrap
    self.run()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 119, in worker
    result = (True, func(*args, **kwds))
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 59, in run_bench_process
    tester.testSignal(a.mSignal, a.mHorizon)
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 163, in testSignal
    self.trainSignal(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 158, in trainSignal
    self.trainModel(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 146, in trainModel
    self.reportModelInfo(lAutoF1);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 137, in reportModelInfo
    iModel.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/ForecastEngine.py", line 40, in getModelInfo
    return  self.mSignalDecomposition.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/SignalDecomposition.py", line 495, in getModelInfo
    return self.mBestModel.getInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/TimeSeriesModel.py", line 85, in getInfo
    logger.info("TIME_DETAIL " + self.mTrend.mTimeInfo.info());
Message: "TIME_DETAIL TimeVariable='Date' TimeMin=0 TimeMax=8 TimeDelta=1.0 Estimation = (0 , 9) Validation = (0 , 9) Test = (0 , 9) Horizon=6"
Arguments: ()
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.5/logging/__init__.py", line 983, in emit
    stream.write(msg)
ValueError: I/O operation on closed file.
Call stack:
  File "/usr/local/bin/ipython3", line 11, in <module>
    sys.exit(start_ipython())
  File "/usr/local/lib/python3.5/dist-packages/IPython/__init__.py", line 119, in start_ipython
    return launch_new_instance(argv=argv, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 595, in launch_instance
    app.initialize(argv)
  File "<decorator-gen-111>", line 2, in initialize
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 74, in catch_config_error
    return method(app, *args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/IPython/terminal/ipapp.py", line 311, in initialize
    self.init_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 298, in init_code
    self._run_cmd_line_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 423, in _run_cmd_line_code
    self._exec_file(fname, shell_futures=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 353, in _exec_file
    raise_exceptions=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/interactiveshell.py", line 2493, in safe_execfile
    self.compile if kw['shell_futures'] else None)
  File "/usr/local/lib/python3.5/dist-packages/IPython/utils/py3compat.py", line 185, in execfile
    exec(compiler(f.read(), fname, 'exec'), glob, loc)
  File "/home/antoine/dev/python/packages/pyaf/benches/run_benchmark_M4Comp.py", line 17, in <module>
    tester.run_multiprocessed();
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 335, in run_multiprocessed
    pool = mp.Pool(processes=nbprocesses, maxtasksperchild=None)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 118, in Pool
    context=self.get_context())
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 168, in __init__
    self._repopulate_pool()
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 233, in _repopulate_pool
    w.start()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 105, in start
    self._popen = self._Popen(self)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 267, in _Popen
    return Popen(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 20, in __init__
    self._launch(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 74, in _launch
    code = process_obj._bootstrap()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 249, in _bootstrap
    self.run()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 119, in worker
    result = (True, func(*args, **kwds))
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 59, in run_bench_process
    tester.testSignal(a.mSignal, a.mHorizon)
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 163, in testSignal
    self.trainSignal(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 158, in trainSignal
    self.trainModel(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 146, in trainModel
    self.reportModelInfo(lAutoF1);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 137, in reportModelInfo
    iModel.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/ForecastEngine.py", line 40, in getModelInfo
    return  self.mSignalDecomposition.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/SignalDecomposition.py", line 495, in getModelInfo
    return self.mBestModel.getInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/TimeSeriesModel.py", line 86, in getInfo
    logger.info("SIGNAL_DETAIL " + self.info());
Message: "SIGNAL_DETAIL SignalVariable='_ECON1880' Min=43.73008 Max=297.53  Mean=129.830952222 StdDev=83.6855578352"
Arguments: ()
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.5/logging/__init__.py", line 983, in emit
    stream.write(msg)
ValueError: I/O operation on closed file.
Call stack:
  File "/usr/local/bin/ipython3", line 11, in <module>
    sys.exit(start_ipython())
  File "/usr/local/lib/python3.5/dist-packages/IPython/__init__.py", line 119, in start_ipython
    return launch_new_instance(argv=argv, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 595, in launch_instance
    app.initialize(argv)
  File "<decorator-gen-111>", line 2, in initialize
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 74, in catch_config_error
    return method(app, *args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/IPython/terminal/ipapp.py", line 311, in initialize
    self.init_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 298, in init_code
    self._run_cmd_line_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 423, in _run_cmd_line_code
    self._exec_file(fname, shell_futures=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 353, in _exec_file
    raise_exceptions=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/interactiveshell.py", line 2493, in safe_execfile
    self.compile if kw['shell_futures'] else None)
  File "/usr/local/lib/python3.5/dist-packages/IPython/utils/py3compat.py", line 185, in execfile
    exec(compiler(f.read(), fname, 'exec'), glob, loc)
  File "/home/antoine/dev/python/packages/pyaf/benches/run_benchmark_M4Comp.py", line 17, in <module>
    tester.run_multiprocessed();
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 335, in run_multiprocessed
    pool = mp.Pool(processes=nbprocesses, maxtasksperchild=None)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 118, in Pool
    context=self.get_context())
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 168, in __init__
    self._repopulate_pool()
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 233, in _repopulate_pool
    w.start()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 105, in start
    self._popen = self._Popen(self)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 267, in _Popen
    return Popen(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 20, in __init__
    self._launch(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 74, in _launch
    code = process_obj._bootstrap()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 249, in _bootstrap
    self.run()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 119, in worker
    result = (True, func(*args, **kwds))
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 59, in run_bench_process
    tester.testSignal(a.mSignal, a.mHorizon)
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 163, in testSignal
    self.trainSignal(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 158, in trainSignal
    self.trainModel(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 146, in trainModel
    self.reportModelInfo(lAutoF1);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 137, in reportModelInfo
    iModel.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/ForecastEngine.py", line 40, in getModelInfo
    return  self.mSignalDecomposition.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/SignalDecomposition.py", line 495, in getModelInfo
    return self.mBestModel.getInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/TimeSeriesModel.py", line 87, in getInfo
    logger.info("BEST_TRANSOFORMATION_TYPE '" + self.mTransformation.get_name("") + "'");
Message: "BEST_TRANSOFORMATION_TYPE '_'"
Arguments: ()
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.5/logging/__init__.py", line 983, in emit
    stream.write(msg)
ValueError: I/O operation on closed file.
Call stack:
  File "/usr/local/bin/ipython3", line 11, in <module>
    sys.exit(start_ipython())
  File "/usr/local/lib/python3.5/dist-packages/IPython/__init__.py", line 119, in start_ipython
    return launch_new_instance(argv=argv, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 595, in launch_instance
    app.initialize(argv)
  File "<decorator-gen-111>", line 2, in initialize
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 74, in catch_config_error
    return method(app, *args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/IPython/terminal/ipapp.py", line 311, in initialize
    self.init_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 298, in init_code
    self._run_cmd_line_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 423, in _run_cmd_line_code
    self._exec_file(fname, shell_futures=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 353, in _exec_file
    raise_exceptions=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/interactiveshell.py", line 2493, in safe_execfile
    self.compile if kw['shell_futures'] else None)
  File "/usr/local/lib/python3.5/dist-packages/IPython/utils/py3compat.py", line 185, in execfile
    exec(compiler(f.read(), fname, 'exec'), glob, loc)
  File "/home/antoine/dev/python/packages/pyaf/benches/run_benchmark_M4Comp.py", line 17, in <module>
    tester.run_multiprocessed();
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 335, in run_multiprocessed
    pool = mp.Pool(processes=nbprocesses, maxtasksperchild=None)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 118, in Pool
    context=self.get_context())
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 168, in __init__
    self._repopulate_pool()
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 233, in _repopulate_pool
    w.start()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 105, in start
    self._popen = self._Popen(self)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 267, in _Popen
    return Popen(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 20, in __init__
    self._launch(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 74, in _launch
    code = process_obj._bootstrap()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 249, in _bootstrap
    self.run()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 119, in worker
    result = (True, func(*args, **kwds))
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 59, in run_bench_process
    tester.testSignal(a.mSignal, a.mHorizon)
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 163, in testSignal
    self.trainSignal(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 158, in trainSignal
    self.trainModel(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 146, in trainModel
    self.reportModelInfo(lAutoF1);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 137, in reportModelInfo
    iModel.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/ForecastEngine.py", line 40, in getModelInfo
    return  self.mSignalDecomposition.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/SignalDecomposition.py", line 495, in getModelInfo
    return self.mBestModel.getInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/TimeSeriesModel.py", line 88, in getInfo
    logger.info("BEST_DECOMPOSITION  '" + self.mOutName + "' [" + self.getFormula() + "]");
Message: "BEST_DECOMPOSITION  '_ECON1880_Lag1Trend_residue_zeroCycle_residue_NoAR' [Lag1Trend + NoCycle + NoAR]"
Arguments: ()
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.5/logging/__init__.py", line 983, in emit
    stream.write(msg)
ValueError: I/O operation on closed file.
Call stack:
  File "/usr/local/bin/ipython3", line 11, in <module>
    sys.exit(start_ipython())
  File "/usr/local/lib/python3.5/dist-packages/IPython/__init__.py", line 119, in start_ipython
    return launch_new_instance(argv=argv, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 595, in launch_instance
    app.initialize(argv)
  File "<decorator-gen-111>", line 2, in initialize
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 74, in catch_config_error
    return method(app, *args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/IPython/terminal/ipapp.py", line 311, in initialize
    self.init_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 298, in init_code
    self._run_cmd_line_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 423, in _run_cmd_line_code
    self._exec_file(fname, shell_futures=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 353, in _exec_file
    raise_exceptions=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/interactiveshell.py", line 2493, in safe_execfile
    self.compile if kw['shell_futures'] else None)
  File "/usr/local/lib/python3.5/dist-packages/IPython/utils/py3compat.py", line 185, in execfile
    exec(compiler(f.read(), fname, 'exec'), glob, loc)
  File "/home/antoine/dev/python/packages/pyaf/benches/run_benchmark_M4Comp.py", line 17, in <module>
    tester.run_multiprocessed();
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 335, in run_multiprocessed
    pool = mp.Pool(processes=nbprocesses, maxtasksperchild=None)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 118, in Pool
    context=self.get_context())
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 168, in __init__
    self._repopulate_pool()
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 233, in _repopulate_pool
    w.start()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 105, in start
    self._popen = self._Popen(self)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 267, in _Popen
    return Popen(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 20, in __init__
    self._launch(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 74, in _launch
    code = process_obj._bootstrap()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 249, in _bootstrap
    self.run()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 119, in worker
    result = (True, func(*args, **kwds))
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 59, in run_bench_process
    tester.testSignal(a.mSignal, a.mHorizon)
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 163, in testSignal
    self.trainSignal(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 158, in trainSignal
    self.trainModel(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 146, in trainModel
    self.reportModelInfo(lAutoF1);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 137, in reportModelInfo
    iModel.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/ForecastEngine.py", line 40, in getModelInfo
    return  self.mSignalDecomposition.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/SignalDecomposition.py", line 495, in getModelInfo
    return self.mBestModel.getInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/TimeSeriesModel.py", line 89, in getInfo
    logger.info("TREND_DETAIL '" + self.mTrend.mOutName + "' [" + self.mTrend.mFormula + "]");
Message: "TREND_DETAIL '_ECON1880_Lag1Trend' [Lag1Trend]"
Arguments: ()
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.5/logging/__init__.py", line 983, in emit
    stream.write(msg)
ValueError: I/O operation on closed file.
Call stack:
  File "/usr/local/bin/ipython3", line 11, in <module>
    sys.exit(start_ipython())
  File "/usr/local/lib/python3.5/dist-packages/IPython/__init__.py", line 119, in start_ipython
    return launch_new_instance(argv=argv, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 595, in launch_instance
    app.initialize(argv)
  File "<decorator-gen-111>", line 2, in initialize
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 74, in catch_config_error
    return method(app, *args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/IPython/terminal/ipapp.py", line 311, in initialize
    self.init_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 298, in init_code
    self._run_cmd_line_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 423, in _run_cmd_line_code
    self._exec_file(fname, shell_futures=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 353, in _exec_file
    raise_exceptions=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/interactiveshell.py", line 2493, in safe_execfile
    self.compile if kw['shell_futures'] else None)
  File "/usr/local/lib/python3.5/dist-packages/IPython/utils/py3compat.py", line 185, in execfile
    exec(compiler(f.read(), fname, 'exec'), glob, loc)
  File "/home/antoine/dev/python/packages/pyaf/benches/run_benchmark_M4Comp.py", line 17, in <module>
    tester.run_multiprocessed();
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 335, in run_multiprocessed
    pool = mp.Pool(processes=nbprocesses, maxtasksperchild=None)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 118, in Pool
    context=self.get_context())
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 168, in __init__
    self._repopulate_pool()
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 233, in _repopulate_pool
    w.start()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 105, in start
    self._popen = self._Popen(self)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 267, in _Popen
    return Popen(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 20, in __init__
    self._launch(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 74, in _launch
    code = process_obj._bootstrap()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 249, in _bootstrap
    self.run()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 119, in worker
    result = (True, func(*args, **kwds))
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 59, in run_bench_process
    tester.testSignal(a.mSignal, a.mHorizon)
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 163, in testSignal
    self.trainSignal(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 158, in trainSignal
    self.trainModel(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 146, in trainModel
    self.reportModelInfo(lAutoF1);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 137, in reportModelInfo
    iModel.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/ForecastEngine.py", line 40, in getModelInfo
    return  self.mSignalDecomposition.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/SignalDecomposition.py", line 495, in getModelInfo
    return self.mBestModel.getInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/TimeSeriesModel.py", line 90, in getInfo
    logger.info("CYCLE_DETAIL '"+ self.mCycle.mOutName + "' [" + self.mCycle.mFormula + "]");
Message: "CYCLE_DETAIL '_ECON1880_Lag1Trend_residue_zeroCycle' [NoCycle]"
Arguments: ()
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.5/logging/__init__.py", line 983, in emit
    stream.write(msg)
ValueError: I/O operation on closed file.
Call stack:
  File "/usr/local/bin/ipython3", line 11, in <module>
    sys.exit(start_ipython())
  File "/usr/local/lib/python3.5/dist-packages/IPython/__init__.py", line 119, in start_ipython
    return launch_new_instance(argv=argv, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 595, in launch_instance
    app.initialize(argv)
  File "<decorator-gen-111>", line 2, in initialize
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 74, in catch_config_error
    return method(app, *args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/IPython/terminal/ipapp.py", line 311, in initialize
    self.init_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 298, in init_code
    self._run_cmd_line_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 423, in _run_cmd_line_code
    self._exec_file(fname, shell_futures=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 353, in _exec_file
    raise_exceptions=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/interactiveshell.py", line 2493, in safe_execfile
    self.compile if kw['shell_futures'] else None)
  File "/usr/local/lib/python3.5/dist-packages/IPython/utils/py3compat.py", line 185, in execfile
    exec(compiler(f.read(), fname, 'exec'), glob, loc)
  File "/home/antoine/dev/python/packages/pyaf/benches/run_benchmark_M4Comp.py", line 17, in <module>
    tester.run_multiprocessed();
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 335, in run_multiprocessed
    pool = mp.Pool(processes=nbprocesses, maxtasksperchild=None)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 118, in Pool
    context=self.get_context())
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 168, in __init__
    self._repopulate_pool()
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 233, in _repopulate_pool
    w.start()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 105, in start
    self._popen = self._Popen(self)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 267, in _Popen
    return Popen(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 20, in __init__
    self._launch(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 74, in _launch
    code = process_obj._bootstrap()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 249, in _bootstrap
    self.run()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 119, in worker
    result = (True, func(*args, **kwds))
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 59, in run_bench_process
    tester.testSignal(a.mSignal, a.mHorizon)
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 163, in testSignal
    self.trainSignal(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 158, in trainSignal
    self.trainModel(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 146, in trainModel
    self.reportModelInfo(lAutoF1);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 137, in reportModelInfo
    iModel.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/ForecastEngine.py", line 40, in getModelInfo
    return  self.mSignalDecomposition.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/SignalDecomposition.py", line 495, in getModelInfo
    return self.mBestModel.getInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/TimeSeriesModel.py", line 91, in getInfo
    logger.info("AUTOREG_DETAIL '" + self.mAR.mOutName + "' [" + self.mAR.mFormula + "]");
Message: "AUTOREG_DETAIL '_ECON1880_Lag1Trend_residue_zeroCycle_residue_NoAR' [NoAR]"
Arguments: ()
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.5/logging/__init__.py", line 983, in emit
    stream.write(msg)
ValueError: I/O operation on closed file.
Call stack:
  File "/usr/local/bin/ipython3", line 11, in <module>
    sys.exit(start_ipython())
  File "/usr/local/lib/python3.5/dist-packages/IPython/__init__.py", line 119, in start_ipython
    return launch_new_instance(argv=argv, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 595, in launch_instance
    app.initialize(argv)
  File "<decorator-gen-111>", line 2, in initialize
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 74, in catch_config_error
    return method(app, *args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/IPython/terminal/ipapp.py", line 311, in initialize
    self.init_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 298, in init_code
    self._run_cmd_line_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 423, in _run_cmd_line_code
    self._exec_file(fname, shell_futures=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 353, in _exec_file
    raise_exceptions=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/interactiveshell.py", line 2493, in safe_execfile
    self.compile if kw['shell_futures'] else None)
  File "/usr/local/lib/python3.5/dist-packages/IPython/utils/py3compat.py", line 185, in execfile
    exec(compiler(f.read(), fname, 'exec'), glob, loc)
  File "/home/antoine/dev/python/packages/pyaf/benches/run_benchmark_M4Comp.py", line 17, in <module>
    tester.run_multiprocessed();
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 335, in run_multiprocessed
    pool = mp.Pool(processes=nbprocesses, maxtasksperchild=None)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 118, in Pool
    context=self.get_context())
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 168, in __init__
    self._repopulate_pool()
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 233, in _repopulate_pool
    w.start()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 105, in start
    self._popen = self._Popen(self)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 267, in _Popen
    return Popen(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 20, in __init__
    self._launch(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 74, in _launch
    code = process_obj._bootstrap()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 249, in _bootstrap
    self.run()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 119, in worker
    result = (True, func(*args, **kwds))
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 59, in run_bench_process
    tester.testSignal(a.mSignal, a.mHorizon)
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 163, in testSignal
    self.trainSignal(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 158, in trainSignal
    self.trainModel(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 146, in trainModel
    self.reportModelInfo(lAutoF1);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 137, in reportModelInfo
    iModel.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/ForecastEngine.py", line 40, in getModelInfo
    return  self.mSignalDecomposition.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/SignalDecomposition.py", line 495, in getModelInfo
    return self.mBestModel.getInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/TimeSeriesModel.py", line 92, in getInfo
    logger.info("MODEL_MAPE MAPE_Fit=" + str(self.mFitPerf.mMAPE) + " MAPE_Forecast=" + str(self.mForecastPerf.mMAPE)  + " MAPE_Test=" + str(self.mTestPerf.mMAPE) );
Message: 'MODEL_MAPE MAPE_Fit=0.225 MAPE_Forecast=0.225 MAPE_Test=0.225'
Arguments: ()
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.5/logging/__init__.py", line 983, in emit
    stream.write(msg)
ValueError: I/O operation on closed file.
Call stack:
  File "/usr/local/bin/ipython3", line 11, in <module>
    sys.exit(start_ipython())
  File "/usr/local/lib/python3.5/dist-packages/IPython/__init__.py", line 119, in start_ipython
    return launch_new_instance(argv=argv, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 595, in launch_instance
    app.initialize(argv)
  File "<decorator-gen-111>", line 2, in initialize
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 74, in catch_config_error
    return method(app, *args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/IPython/terminal/ipapp.py", line 311, in initialize
    self.init_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 298, in init_code
    self._run_cmd_line_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 423, in _run_cmd_line_code
    self._exec_file(fname, shell_futures=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 353, in _exec_file
    raise_exceptions=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/interactiveshell.py", line 2493, in safe_execfile
    self.compile if kw['shell_futures'] else None)
  File "/usr/local/lib/python3.5/dist-packages/IPython/utils/py3compat.py", line 185, in execfile
    exec(compiler(f.read(), fname, 'exec'), glob, loc)
  File "/home/antoine/dev/python/packages/pyaf/benches/run_benchmark_M4Comp.py", line 17, in <module>
    tester.run_multiprocessed();
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 335, in run_multiprocessed
    pool = mp.Pool(processes=nbprocesses, maxtasksperchild=None)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 118, in Pool
    context=self.get_context())
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 168, in __init__
    self._repopulate_pool()
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 233, in _repopulate_pool
    w.start()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 105, in start
    self._popen = self._Popen(self)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 267, in _Popen
    return Popen(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 20, in __init__
    self._launch(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 74, in _launch
    code = process_obj._bootstrap()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 249, in _bootstrap
    self.run()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 119, in worker
    result = (True, func(*args, **kwds))
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 59, in run_bench_process
    tester.testSignal(a.mSignal, a.mHorizon)
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 163, in testSignal
    self.trainSignal(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 158, in trainSignal
    self.trainModel(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 146, in trainModel
    self.reportModelInfo(lAutoF1);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 137, in reportModelInfo
    iModel.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/ForecastEngine.py", line 40, in getModelInfo
    return  self.mSignalDecomposition.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/SignalDecomposition.py", line 495, in getModelInfo
    return self.mBestModel.getInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/TimeSeriesModel.py", line 93, in getInfo
    logger.info("MODEL_MASE MASE_Fit=" + str(self.mFitPerf.mMASE) + " MASE_Forecast=" + str(self.mForecastPerf.mMASE)  + " MASE_Test=" + str(self.mTestPerf.mMASE) );
Message: 'MODEL_MASE MASE_Fit=0.8889 MASE_Forecast=0.8889 MASE_Test=0.8889'
Arguments: ()
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.5/logging/__init__.py", line 983, in emit
    stream.write(msg)
ValueError: I/O operation on closed file.
Call stack:
  File "/usr/local/bin/ipython3", line 11, in <module>
    sys.exit(start_ipython())
  File "/usr/local/lib/python3.5/dist-packages/IPython/__init__.py", line 119, in start_ipython
    return launch_new_instance(argv=argv, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 595, in launch_instance
    app.initialize(argv)
  File "<decorator-gen-111>", line 2, in initialize
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 74, in catch_config_error
    return method(app, *args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/IPython/terminal/ipapp.py", line 311, in initialize
    self.init_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 298, in init_code
    self._run_cmd_line_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 423, in _run_cmd_line_code
    self._exec_file(fname, shell_futures=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 353, in _exec_file
    raise_exceptions=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/interactiveshell.py", line 2493, in safe_execfile
    self.compile if kw['shell_futures'] else None)
  File "/usr/local/lib/python3.5/dist-packages/IPython/utils/py3compat.py", line 185, in execfile
    exec(compiler(f.read(), fname, 'exec'), glob, loc)
  File "/home/antoine/dev/python/packages/pyaf/benches/run_benchmark_M4Comp.py", line 17, in <module>
    tester.run_multiprocessed();
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 335, in run_multiprocessed
    pool = mp.Pool(processes=nbprocesses, maxtasksperchild=None)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 118, in Pool
    context=self.get_context())
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 168, in __init__
    self._repopulate_pool()
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 233, in _repopulate_pool
    w.start()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 105, in start
    self._popen = self._Popen(self)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 267, in _Popen
    return Popen(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 20, in __init__
    self._launch(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 74, in _launch
    code = process_obj._bootstrap()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 249, in _bootstrap
    self.run()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 119, in worker
    result = (True, func(*args, **kwds))
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 59, in run_bench_process
    tester.testSignal(a.mSignal, a.mHorizon)
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 163, in testSignal
    self.trainSignal(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 158, in trainSignal
    self.trainModel(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 146, in trainModel
    self.reportModelInfo(lAutoF1);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 137, in reportModelInfo
    iModel.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/ForecastEngine.py", line 40, in getModelInfo
    return  self.mSignalDecomposition.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/SignalDecomposition.py", line 495, in getModelInfo
    return self.mBestModel.getInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/TimeSeriesModel.py", line 94, in getInfo
    logger.info("MODEL_L1 L1_Fit=" + str(self.mFitPerf.mL2) + " L1_Forecast=" + str(self.mForecastPerf.mL2)  + " L1_Test=" + str(self.mTestPerf.mL1) );
Message: 'MODEL_L1 L1_Fit=59.7867317182 L1_Forecast=59.7867317182 L1_Test=36.6367066667'
Arguments: ()
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.5/logging/__init__.py", line 983, in emit
    stream.write(msg)
ValueError: I/O operation on closed file.
Call stack:
  File "/usr/local/bin/ipython3", line 11, in <module>
    sys.exit(start_ipython())
  File "/usr/local/lib/python3.5/dist-packages/IPython/__init__.py", line 119, in start_ipython
    return launch_new_instance(argv=argv, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 595, in launch_instance
    app.initialize(argv)
  File "<decorator-gen-111>", line 2, in initialize
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 74, in catch_config_error
    return method(app, *args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/IPython/terminal/ipapp.py", line 311, in initialize
    self.init_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 298, in init_code
    self._run_cmd_line_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 423, in _run_cmd_line_code
    self._exec_file(fname, shell_futures=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 353, in _exec_file
    raise_exceptions=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/interactiveshell.py", line 2493, in safe_execfile
    self.compile if kw['shell_futures'] else None)
  File "/usr/local/lib/python3.5/dist-packages/IPython/utils/py3compat.py", line 185, in execfile
    exec(compiler(f.read(), fname, 'exec'), glob, loc)
  File "/home/antoine/dev/python/packages/pyaf/benches/run_benchmark_M4Comp.py", line 17, in <module>
    tester.run_multiprocessed();
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 335, in run_multiprocessed
    pool = mp.Pool(processes=nbprocesses, maxtasksperchild=None)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 118, in Pool
    context=self.get_context())
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 168, in __init__
    self._repopulate_pool()
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 233, in _repopulate_pool
    w.start()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 105, in start
    self._popen = self._Popen(self)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 267, in _Popen
    return Popen(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 20, in __init__
    self._launch(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 74, in _launch
    code = process_obj._bootstrap()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 249, in _bootstrap
    self.run()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 119, in worker
    result = (True, func(*args, **kwds))
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 59, in run_bench_process
    tester.testSignal(a.mSignal, a.mHorizon)
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 163, in testSignal
    self.trainSignal(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 158, in trainSignal
    self.trainModel(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 146, in trainModel
    self.reportModelInfo(lAutoF1);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 137, in reportModelInfo
    iModel.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/ForecastEngine.py", line 40, in getModelInfo
    return  self.mSignalDecomposition.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/SignalDecomposition.py", line 495, in getModelInfo
    return self.mBestModel.getInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/TimeSeriesModel.py", line 95, in getInfo
    logger.info("MODEL_L2 L2_Fit=" + str(self.mFitPerf.mL2) + " L2_Forecast=" + str(self.mForecastPerf.mL2)  + " L2_Test=" + str(self.mTestPerf.mL2) );
Message: 'MODEL_L2 L2_Fit=59.7867317182 L2_Forecast=59.7867317182 L2_Test=59.7867317182'
Arguments: ()
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.5/logging/__init__.py", line 983, in emit
    stream.write(msg)
ValueError: I/O operation on closed file.
Call stack:
  File "/usr/local/bin/ipython3", line 11, in <module>
    sys.exit(start_ipython())
  File "/usr/local/lib/python3.5/dist-packages/IPython/__init__.py", line 119, in start_ipython
    return launch_new_instance(argv=argv, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 595, in launch_instance
    app.initialize(argv)
  File "<decorator-gen-111>", line 2, in initialize
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 74, in catch_config_error
    return method(app, *args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/IPython/terminal/ipapp.py", line 311, in initialize
    self.init_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 298, in init_code
    self._run_cmd_line_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 423, in _run_cmd_line_code
    self._exec_file(fname, shell_futures=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 353, in _exec_file
    raise_exceptions=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/interactiveshell.py", line 2493, in safe_execfile
    self.compile if kw['shell_futures'] else None)
  File "/usr/local/lib/python3.5/dist-packages/IPython/utils/py3compat.py", line 185, in execfile
    exec(compiler(f.read(), fname, 'exec'), glob, loc)
  File "/home/antoine/dev/python/packages/pyaf/benches/run_benchmark_M4Comp.py", line 17, in <module>
    tester.run_multiprocessed();
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 335, in run_multiprocessed
    pool = mp.Pool(processes=nbprocesses, maxtasksperchild=None)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 118, in Pool
    context=self.get_context())
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 168, in __init__
    self._repopulate_pool()
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 233, in _repopulate_pool
    w.start()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 105, in start
    self._popen = self._Popen(self)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 267, in _Popen
    return Popen(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 20, in __init__
    self._launch(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 74, in _launch
    code = process_obj._bootstrap()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 249, in _bootstrap
    self.run()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 119, in worker
    result = (True, func(*args, **kwds))
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 59, in run_bench_process
    tester.testSignal(a.mSignal, a.mHorizon)
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 163, in testSignal
    self.trainSignal(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 158, in trainSignal
    self.trainModel(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 146, in trainModel
    self.reportModelInfo(lAutoF1);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 137, in reportModelInfo
    iModel.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/ForecastEngine.py", line 40, in getModelInfo
    return  self.mSignalDecomposition.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/SignalDecomposition.py", line 495, in getModelInfo
    return self.mBestModel.getInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/TimeSeriesModel.py", line 96, in getInfo
    logger.info("MODEL_COMPLEXITY " + str(self.getComplexity()) );
Message: 'MODEL_COMPLEXITY 32'
Arguments: ()
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.5/logging/__init__.py", line 983, in emit
    stream.write(msg)
ValueError: I/O operation on closed file.
Call stack:
  File "/usr/local/bin/ipython3", line 11, in <module>
    sys.exit(start_ipython())
  File "/usr/local/lib/python3.5/dist-packages/IPython/__init__.py", line 119, in start_ipython
    return launch_new_instance(argv=argv, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 595, in launch_instance
    app.initialize(argv)
  File "<decorator-gen-111>", line 2, in initialize
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 74, in catch_config_error
    return method(app, *args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/IPython/terminal/ipapp.py", line 311, in initialize
    self.init_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 298, in init_code
    self._run_cmd_line_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 423, in _run_cmd_line_code
    self._exec_file(fname, shell_futures=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 353, in _exec_file
    raise_exceptions=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/interactiveshell.py", line 2493, in safe_execfile
    self.compile if kw['shell_futures'] else None)
  File "/usr/local/lib/python3.5/dist-packages/IPython/utils/py3compat.py", line 185, in execfile
    exec(compiler(f.read(), fname, 'exec'), glob, loc)
  File "/home/antoine/dev/python/packages/pyaf/benches/run_benchmark_M4Comp.py", line 17, in <module>
    tester.run_multiprocessed();
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 335, in run_multiprocessed
    pool = mp.Pool(processes=nbprocesses, maxtasksperchild=None)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 118, in Pool
    context=self.get_context())
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 168, in __init__
    self._repopulate_pool()
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 233, in _repopulate_pool
    w.start()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 105, in start
    self._popen = self._Popen(self)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 267, in _Popen
    return Popen(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 20, in __init__
    self._launch(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 74, in _launch
    code = process_obj._bootstrap()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 249, in _bootstrap
    self.run()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 119, in worker
    result = (True, func(*args, **kwds))
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 59, in run_bench_process
    tester.testSignal(a.mSignal, a.mHorizon)
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 163, in testSignal
    self.trainSignal(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 158, in trainSignal
    self.trainModel(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 146, in trainModel
    self.reportModelInfo(lAutoF1);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 137, in reportModelInfo
    iModel.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/ForecastEngine.py", line 40, in getModelInfo
    return  self.mSignalDecomposition.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/SignalDecomposition.py", line 495, in getModelInfo
    return self.mBestModel.getInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/TimeSeriesModel.py", line 97, in getInfo
    logger.info("AR_MODEL_DETAIL_START");
Message: 'AR_MODEL_DETAIL_START'
Arguments: ()
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.5/logging/__init__.py", line 983, in emit
    stream.write(msg)
ValueError: I/O operation on closed file.
Call stack:
  File "/usr/local/bin/ipython3", line 11, in <module>
    sys.exit(start_ipython())
  File "/usr/local/lib/python3.5/dist-packages/IPython/__init__.py", line 119, in start_ipython
    return launch_new_instance(argv=argv, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 595, in launch_instance
    app.initialize(argv)
  File "<decorator-gen-111>", line 2, in initialize
  File "/usr/local/lib/python3.5/dist-packages/traitlets/config/application.py", line 74, in catch_config_error
    return method(app, *args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/IPython/terminal/ipapp.py", line 311, in initialize
    self.init_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 298, in init_code
    self._run_cmd_line_code()
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 423, in _run_cmd_line_code
    self._exec_file(fname, shell_futures=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/shellapp.py", line 353, in _exec_file
    raise_exceptions=True)
  File "/usr/local/lib/python3.5/dist-packages/IPython/core/interactiveshell.py", line 2493, in safe_execfile
    self.compile if kw['shell_futures'] else None)
  File "/usr/local/lib/python3.5/dist-packages/IPython/utils/py3compat.py", line 185, in execfile
    exec(compiler(f.read(), fname, 'exec'), glob, loc)
  File "/home/antoine/dev/python/packages/pyaf/benches/run_benchmark_M4Comp.py", line 17, in <module>
    tester.run_multiprocessed();
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 335, in run_multiprocessed
    pool = mp.Pool(processes=nbprocesses, maxtasksperchild=None)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 118, in Pool
    context=self.get_context())
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 168, in __init__
    self._repopulate_pool()
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 233, in _repopulate_pool
    w.start()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 105, in start
    self._popen = self._Popen(self)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 267, in _Popen
    return Popen(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 20, in __init__
    self._launch(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 74, in _launch
    code = process_obj._bootstrap()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 249, in _bootstrap
    self.run()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python3.5/multiprocessing/pool.py", line 119, in worker
    result = (True, func(*args, **kwds))
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 59, in run_bench_process
    tester.testSignal(a.mSignal, a.mHorizon)
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 163, in testSignal
    self.trainSignal(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 158, in trainSignal
    self.trainModel(iSignal, iHorizon);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 146, in trainModel
    self.reportModelInfo(lAutoF1);
  File "/home/antoine/dev/python/packages/pyaf/Bench/GenericBenchmark.py", line 137, in reportModelInfo
    iModel.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/ForecastEngine.py", line 40, in getModelInfo
    return  self.mSignalDecomposition.getModelInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/SignalDecomposition.py", line 495, in getModelInfo
    return self.mBestModel.getInfo();
  File "/home/antoine/dev/python/packages/pyaf/TS/TimeSeriesModel.py", line 99, in getInfo
    logger.info("AR_MODEL_DETAIL_END");
Message: 'AR_MODEL_DETAIL_END'
Arguments: ()
      Transformation                                              Model  \
0          _ECON1880  _ECON1880_Lag1Trend_residue_zeroCycle_residue_...   
1          _ECON1880  _ECON1880_Lag1Trend_residue_bestCycle_byL2_res...   
2      Diff_ECON1880  Diff_ECON1880_Lag1Trend_residue_zeroCycle_resi...   
3      Diff_ECON1880  Diff_ECON1880_Lag1Trend_residue_bestCycle_byL2...   
4          _ECON1880  _ECON1880_PolyTrend_residue_zeroCycle_residue_...   
5          _ECON1880  _ECON1880_PolyTrend_residue_bestCycle_byL2_res...   
6      Diff_ECON1880  Diff_ECON1880_PolyTrend_residue_zeroCycle_resi...   
7      Diff_ECON1880  Diff_ECON1880_PolyTrend_residue_bestCycle_byL2...   
8    CumSum_ECON1880  CumSum_ECON1880_Lag1Trend_residue_zeroCycle_re...   
9    CumSum_ECON1880  CumSum_ECON1880_Lag1Trend_residue_bestCycle_by...   
10     Diff_ECON1880  Diff_ECON1880_LinearTrend_residue_zeroCycle_re...   
11     Diff_ECON1880  Diff_ECON1880_LinearTrend_residue_bestCycle_by...   
12         _ECON1880  _ECON1880_LinearTrend_residue_zeroCycle_residu...   
13         _ECON1880  _ECON1880_LinearTrend_residue_bestCycle_byL2_r...   
14     Diff_ECON1880  Diff_ECON1880_ConstantTrend_residue_zeroCycle_...   
15     Diff_ECON1880  Diff_ECON1880_ConstantTrend_residue_bestCycle_...   
16   CumSum_ECON1880  CumSum_ECON1880_PolyTrend_residue_zeroCycle_re...   
17   CumSum_ECON1880  CumSum_ECON1880_PolyTrend_residue_bestCycle_by...   
18  RelDiff_ECON1880  RelDiff_ECON1880_ConstantTrend_residue_zeroCyc...   
19  RelDiff_ECON1880  RelDiff_ECON1880_ConstantTrend_residue_bestCyc...   
20  RelDiff_ECON1880  RelDiff_ECON1880_LinearTrend_residue_zeroCycle...   
21  RelDiff_ECON1880  RelDiff_ECON1880_PolyTrend_residue_zeroCycle_r...   
22  RelDiff_ECON1880  RelDiff_ECON1880_LinearTrend_residue_bestCycle...   
23  RelDiff_ECON1880  RelDiff_ECON1880_PolyTrend_residue_bestCycle_b...   
24  RelDiff_ECON1880  RelDiff_ECON1880_Lag1Trend_residue_zeroCycle_r...   
25  RelDiff_ECON1880  RelDiff_ECON1880_Lag1Trend_residue_bestCycle_b...   
26         _ECON1880  _ECON1880_ConstantTrend_residue_zeroCycle_resi...   
27         _ECON1880  _ECON1880_ConstantTrend_residue_bestCycle_byL2...   
28   CumSum_ECON1880  CumSum_ECON1880_LinearTrend_residue_zeroCycle_...   
29   CumSum_ECON1880  CumSum_ECON1880_LinearTrend_residue_bestCycle_...   
30   CumSum_ECON1880  CumSum_ECON1880_ConstantTrend_residue_zeroCycl...   
31   CumSum_ECON1880  CumSum_ECON1880_ConstantTrend_residue_bestCycl...   

    Complexity  FitCount       FitL1       FitL2  FitMAPE  FitMASE  \
0           32         9   36.636707   59.786732   0.2250   0.8889   
1           40         9   36.636707   59.786732   0.2250   0.8889   
2           64         9   36.636707   59.786732   0.2250   0.8889   
3           72         9   36.636707   59.786732   0.2250   0.8889   
4           16         9   29.432646   38.660616   0.2549   0.7141   
5           24         9   29.432646   38.660616   0.2549   0.7141   
6           48         9   31.556862   39.162819   0.2883   0.7656   
7           56         9   31.556862   39.162819   0.2883   0.7656   
8           64         9   41.495604   62.294667   0.3161   1.0068   
9           72         9   41.495604   62.294667   0.3161   1.0068   
10          48         9   34.148835   41.873206   0.3213   0.8285   
11          56         9   34.148835   41.873206   0.3213   0.8285   
12          16         9   44.370494   55.752840   0.3854   1.0765   
13          24         9   44.370494   55.752840   0.3854   1.0765   
14          32         9   46.792410   53.762407   0.4866   1.1353   
15          40         9   46.792410   53.762407   0.4866   1.1353   
16          48         9   47.516610   67.071688   0.5062   1.1529   
17          56         9   47.516610   67.071688   0.5062   1.1529   
18          32         9   86.100872  120.069283   0.5194   2.0890   
19          40         9   86.100872  120.069283   0.5194   2.0890   
20          48         9   86.100872  120.069283   0.5194   2.0890   
21          48         9   86.100872  120.069283   0.5194   2.0890   
22          56         9   86.100872  120.069283   0.5194   2.0890   
23          56         9   86.100872  120.069283   0.5194   2.0890   
24          64         9   86.100872  120.069283   0.5194   2.0890   
25          72         9   86.100872  120.069283   0.5194   2.0890   
26           0         9   66.376894   83.685558   0.6680   1.6105   
27           8         9   66.376894   83.685558   0.6680   1.6105   
28          48         9   87.954896  118.653784   0.7968   2.1340   
29          56         9   87.954896  118.653784   0.7968   2.1340   
30          32         9  171.337589  207.351716   1.9492   4.1570   
31          40         9  171.337589  207.351716   1.9492   4.1570   

    ForecastCount  ForecastL1  ForecastL2  ForecastMAPE  ForecastMASE  \
0               9   36.636707   59.786732        0.2250        0.8889   
1               9   36.636707   59.786732        0.2250        0.8889   
2               9   36.636707   59.786732        0.2250        0.8889   
3               9   36.636707   59.786732        0.2250        0.8889   
4               9   29.432646   38.660616        0.2549        0.7141   
5               9   29.432646   38.660616        0.2549        0.7141   
6               9   31.556862   39.162819        0.2883        0.7656   
7               9   31.556862   39.162819        0.2883        0.7656   
8               9   41.495604   62.294667        0.3161        1.0068   
9               9   41.495604   62.294667        0.3161        1.0068   
10              9   34.148835   41.873206        0.3213        0.8285   
11              9   34.148835   41.873206        0.3213        0.8285   
12              9   44.370494   55.752840        0.3854        1.0765   
13              9   44.370494   55.752840        0.3854        1.0765   
14              9   46.792410   53.762407        0.4866        1.1353   
15              9   46.792410   53.762407        0.4866        1.1353   
16              9   47.516610   67.071688        0.5062        1.1529   
17              9   47.516610   67.071688        0.5062        1.1529   
18              9   86.100872  120.069283        0.5194        2.0890   
19              9   86.100872  120.069283        0.5194        2.0890   
20              9   86.100872  120.069283        0.5194        2.0890   
21              9   86.100872  120.069283        0.5194        2.0890   
22              9   86.100872  120.069283        0.5194        2.0890   
23              9   86.100872  120.069283        0.5194        2.0890   
24              9   86.100872  120.069283        0.5194        2.0890   
25              9   86.100872  120.069283        0.5194        2.0890   
26              9   66.376894   83.685558        0.6680        1.6105   
27              9   66.376894   83.685558        0.6680        1.6105   
28              9   87.954896  118.653784        0.7968        2.1340   
29              9   87.954896  118.653784        0.7968        2.1340   
30              9  171.337589  207.351716        1.9492        4.1570   
31              9  171.337589  207.351716        1.9492        4.1570   

    TestCount      TestL1      TestL2  TestMAPE  TestMASE  
0           9   36.636707   59.786732    0.2250    0.8889  
1           9   36.636707   59.786732    0.2250    0.8889  
2           9   36.636707   59.786732    0.2250    0.8889  
3           9   36.636707   59.786732    0.2250    0.8889  
4           9   29.432646   38.660616    0.2549    0.7141  
5           9   29.432646   38.660616    0.2549    0.7141  
6           9   31.556862   39.162819    0.2883    0.7656  
7           9   31.556862   39.162819    0.2883    0.7656  
8           9   41.495604   62.294667    0.3161    1.0068  
9           9   41.495604   62.294667    0.3161    1.0068  
10          9   34.148835   41.873206    0.3213    0.8285  
11          9   34.148835   41.873206    0.3213    0.8285  
12          9   44.370494   55.752840    0.3854    1.0765  
13          9   44.370494   55.752840    0.3854    1.0765  
14          9   46.792410   53.762407    0.4866    1.1353  
15          9   46.792410   53.762407    0.4866    1.1353  
16          9   47.516610   67.071688    0.5062    1.1529  
17          9   47.516610   67.071688    0.5062    1.1529  
18          9   86.100872  120.069283    0.5194    2.0890  
19          9   86.100872  120.069283    0.5194    2.0890  
20          9   86.100872  120.069283    0.5194    2.0890  
21          9   86.100872  120.069283    0.5194    2.0890  
22          9   86.100872  120.069283    0.5194    2.0890  
23          9   86.100872  120.069283    0.5194    2.0890  
24          9   86.100872  120.069283    0.5194    2.0890  
25          9   86.100872  120.069283    0.5194    2.0890  
26          9   66.376894   83.685558    0.6680    1.6105  
27          9   66.376894   83.685558    0.6680    1.6105  
28          9   87.954896  118.653784    0.7968    2.1340  
29          9   87.954896  118.653784    0.7968    2.1340  
30          9  171.337589  207.351716    1.9492    4.1570  
31          9  171.337589  207.351716    1.9492    4.1570  
  Transformation                                              Model  \
0      _ECON1880  _ECON1880_Lag1Trend_residue_zeroCycle_residue_...   
1      _ECON1880  _ECON1880_Lag1Trend_residue_bestCycle_byL2_res...   
2  Diff_ECON1880  Diff_ECON1880_Lag1Trend_residue_zeroCycle_resi...   
3  Diff_ECON1880  Diff_ECON1880_Lag1Trend_residue_bestCycle_byL2...   
4      _ECON1880  _ECON1880_PolyTrend_residue_zeroCycle_residue_...   

   Complexity  FitCount      FitL1      FitL2  FitMAPE  FitMASE  \
0          32         9  36.636707  59.786732   0.2250   0.8889   
1          40         9  36.636707  59.786732   0.2250   0.8889   
2          64         9  36.636707  59.786732   0.2250   0.8889   
3          72         9  36.636707  59.786732   0.2250   0.8889   
4          16         9  29.432646  38.660616   0.2549   0.7141   

   ForecastCount  ForecastL1  ForecastL2  ForecastMAPE  ForecastMASE  \
0              9   36.636707   59.786732        0.2250        0.8889   
1              9   36.636707   59.786732        0.2250        0.8889   
2              9   36.636707   59.786732        0.2250        0.8889   
3              9   36.636707   59.786732        0.2250        0.8889   
4              9   29.432646   38.660616        0.2549        0.7141   

   TestCount     TestL1     TestL2  TestMAPE  TestMASE  
0          9  36.636707  59.786732    0.2250    0.8889  
1          9  36.636707  59.786732    0.2250    0.8889  
2          9  36.636707  59.786732    0.2250    0.8889  
3          9  36.636707  59.786732    0.2250    0.8889  
4          9  29.432646  38.660616    0.2549    0.7141  
6 0    288.77709
1    302.80242
2    318.61969
3    318.94706
4    311.31412
5    369.91120
Name: ECON1880, dtype: float64
6 0    257.38378
1    257.38378
2    257.38378
3    257.38378
4    257.38378
5    257.38378
Name: ECON1880_Forecast, dtype: float64
FORECAST_DETAIL_ACTUAL ECON1880 ECON1880 [ 288.77709  302.80242  318.61969  318.94706  311.31412  369.9112 ]
FORECAST_DETAIL_PREDICTED ECON1880 ECON1880 [ 257.38378  257.38378  257.38378  257.38378  257.38378  257.38378]
BENCHMARK_PERF_DETAIL ECON1880 ECON1880 9 6 5.883883237838745 Lag1Trend + NoCycle + NoAR 6 0.1869 0.2086 3.1645 61.0114833333 66.0292469472 -5.83942370046
BENCHMARK_SUCCESS 'M4_COMP_ECONOMICS_ECON1880_6'
